packages/orchestrator/internal/server/sandboxes.go

```diff
@@ -430,42 +388,27 @@ func (s *server) Pause(ctx context.Context, in *orchestrator.SandboxPauseRequest
 		}
 	}(context.WithoutCancel(ctx))
 
-	sandboxLifeCycleEventsWriteFlag, flagErr := s.featureFlags.BoolFlag(
-		featureflags.SandboxLifeCycleEventsWriteFlagName, sbx.Runtime.SandboxID)
-	if flagErr != nil {
-		zap.L().Error("soft failing during sandbox lifecycle events write feature flag receive", zap.Error(flagErr))
+	teamID, err := uuid.Parse(sbx.Runtime.TeamID)
+	if err != nil {
+		sbxlogger.I(sbx).Error("error parsing team ID", zap.String("team_id", sbx.Runtime.TeamID), zap.Error(err))
 	}
 
-	if sandboxLifeCycleEventsWriteFlag {
-		go func() {
-			buildId := ""
-			if sbx.APIStoredConfig != nil {
-				buildId = sbx.APIStoredConfig.BuildId
-			}
-
-			teamID, err := uuid.Parse(sbx.Runtime.TeamID)
-			if err != nil {
-				sbxlogger.I(sbx).Error("error parsing team ID", zap.String("team_id", sbx.Runtime.TeamID), zap.Error(err))
-				return
-			}
-
-			err = s.sandboxEventBatcher.Push(clickhouse.SandboxEvent{
-				Timestamp:          time.Now().UTC(),
-				SandboxID:          sbx.Runtime.SandboxID,
-				SandboxTemplateID:  sbx.Config.BaseTemplateID,
-				SandboxBuildID:     buildId,
-				SandboxTeamID:      teamID,
-				SandboxExecutionID: sbx.Runtime.ExecutionID,
-				EventCategory:      string(clickhouse.SandboxEventCategoryLifecycle),
-				EventLabel:         string(clickhouse.SandboxEventLabelPause),
-				EventData:          sql.NullString{String: "", Valid: false},
-			})
-			if err != nil {
-				sbxlogger.I(sbx).Error(
-					"error inserting sandbox lifecycle event", zap.String("event_label", string(clickhouse.SandboxEventLabelPause)), zap.Error(err))
-			}
-		}()
+	buildId := ""
+	if sbx.APIStoredConfig != nil {
+		buildId = sbx.APIStoredConfig.BuildId
 	}
 
+	go s.sbxEventsService.HandleEvent(ctx, event.SandboxEvent{
```

Comment:
alternately, we could refactor the `HandleEvent` function to avoid the `ctx` parameter. If not though, this needs a `context.WithoutCancel`